src/WindowsTracing/include/WindowsTracing/Tracer.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WINDOWS_TRACING_TRACER_H_
+#define WINDOWS_TRACING_TRACER_H_
+
+#include <memory>
+
+#include "WindowsTracing/TracerInterface.h"
+#include "WindowsTracing/TracerListener.h"
+#include "capture.pb.h"
+
+namespace orbit_windows_tracing {
+
+// Forwards interesting Windows kernel events to a TraceListener.
+class Tracer : public TracerInterface {
```

Comment:
So again on this class, this is still strange to me.
My understanding is that you have `TracerInterface` so that `KrabsTracer` doesn't have to be exposed in the `include` directory and can be instantiated in the `.cpp` file. So far so good.
But you also have `Tracer` implement this interface and that is what I'm confused about: You have a `TracerInterface` (`Tracer`) contain another `TracerInterface` (`KrabsTracer`), which is a bit strange; And `TracerInterface` now has two usages, one exposing `Tracer` as `TracerInterface`, one hiding `KrabsTracer` behind `TracerInterface`.
As I don't currently see a reason for `Tracer` to implement `TracerInterface`, only `KrabsTracer` should implement it.

So in the end I think that `public TracerInterface` and the two `override`s would simply be removed.
As a result, `TracerInterface` would be internal to the module and should end up in `orbit_windows_tracing_internal` as it's in the `include` directory, again as per https://github.com/google/orbit/blob/main/DEVELOPMENT.md#namespaces

Reply:
+1. Basically simply drop the inheritance of TracerInterface here and that's it.

Reply:
Both `Tracer` and `KrabsTracer` need to store the CaptureOptions and TracerListener. Having them both implement the `TracerInterface` puts that logic at one place. Also, both can be used interchangeably through the `TracerInterface`. 

Removing the inheritance in Tracer would mean duplicating code for storing `CaptureOptions` and `TracerListener`, and losing the property that `Tracer` could be thought of as just an implementation of the `TracerInterface`, which I find nice as a user as the interface acts as sort of lean documentation. Also, we could imagine exposing different implementations of the TracerInterface publicly, in which case we need a public interface.

Reply:
@florian-kuebler , ok I ended up removing the TracerInterface in favor of a TracerImpl.h/.cpp the hierarchy, ptal.